src/screens/ShareTrip/ShareTrip.tsx

```diff
@@ -23,26 +34,145 @@ type Props = {
 };
 
 export default function ShareTrip(props: Props): JSX.Element {
+    console.log(props.route);
     const { t } = useTranslation();
-    const trip = props.route.params.trip;
+    const [mode, setMode] = useState<Mode>(
+        props.route.params.tripId ? Mode.SHARE_TRIP : Mode.JOIN_TRIP
+    );
 
-    const [execute, { loading }] = useCreateInvitationMutation();
-    const getInvitationLinkPromise = (): Promise<FetchResult<CreateInvitationMutation>> => {
-        return execute({
-            variables: {
-                input: {
-                    tripId: trip.id,
+    useEffect(() => {
+        async function getInitialUrl(): Promise<void> {
+            const link = await Linking.getInitialURL();
+            if (link && link !== null) {
+                console.log("we got a initial URL link ded");
+                console.log(link);
+            }
+        }
+        getInitialUrl();
+        Linking.addEventListener("url", (event) => {
+            console.log("got event listener callback");
+            console.log(event);
+        });
+    }, []);
+
+    const [executeCreateInvitation, { loading: loadingCreateInvitation }] =
+        useCreateInvitationMutation();
+    const [executeJoinTripMutation, { data: joinTripData, loading: loadingJoinTrip }] =
+        useJoinTripMutation();
+
+    const [executeGetTripQuery, { data: trip, error: getTripError, called: tripQueryCalled }] =
+        useGetTripLazyQuery();
+
+    const [
+        executeGetInvitation,
+        { data: invitation, error: getInvitationError, called: invitationQueryCalled },
+    ] = useGetInvitationLazyQuery({
+        notifyOnNetworkStatusChange: true,
+        fetchPolicy: "network-only",
+        onCompleted: (data) => {
+            console.log("fetching data onCompleted");
```

Comment:
Please see comments above regarding log output

Reply:
I've removed as i think we don't need this output, we've just added it for testing purposes to check if this function is called